File: frontend/src/types/execution.ts
export interface Execution {
  id: string
  workflow_id: string
  workflow_name?: string
  folder_id?: string
  folder_path?: string   // Örn: "Satış > Günlük"
  status: 'pending' | 'running' | 'success' | 'failed' | 'cancelled'
  trigger_type: 'manual' | 'scheduled' | 'chained'
  error_message?: string
  rows_processed: number
  rows_failed: number
  started_at?: string
  finished_at?: string
  created_at: string
}

export interface ExecutionLog {
  id: number
  execution_id: string
  node_id?: string
  level: 'debug' | 'info' | 'warning' | 'error'
  message: string
  created_at: string
}

export interface ExecutionDetail extends Execution {
  logs: ExecutionLog[]
}

export interface TimelineNodeEntry {
  node_id: string
  node_label: string
  start_time: string
  end_time: string
  duration_seconds: number
  status: 'success' | 'failed'
  row_count: number
}

export interface ExecutionTimeline {
  execution_id: string
  started_at?: string
  finished_at?: string
  total_duration_seconds: number
  nodes: TimelineNodeEntry[]
}

File: backend/app/schemas/execution.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ExecutionResponse(BaseModel):
    id: str
    workflow_id: str
    workflow_name: Optional[str] = None   # JOIN ile doldurulur
    folder_id: Optional[str] = None       # Workflow'un klasörü
    folder_path: Optional[str] = None     # Üst > Alt klasör yolu (örn: "Satış > Günlük")
    status: str
    trigger_type: str
    error_message: Optional[str] = None
    rows_processed: int
    rows_failed: int
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ExecutionLogResponse(BaseModel):
    id: int
    execution_id: str
    node_id: Optional[str] = None
    level: str
    message: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ExecutionDetail(ExecutionResponse):
    logs: list[ExecutionLogResponse] = []


class TimelineNodeEntry(BaseModel):
    node_id: str
    node_label: str
    start_time: datetime
    end_time: datetime
    duration_seconds: float
    status: str  # success | failed
    row_count: int


class ExecutionTimeline(BaseModel):
    execution_id: str
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    total_duration_seconds: float = 0
    nodes: list[TimelineNodeEntry] = []
